File: src/main/java/historywowa/domain/member/presentation/dto/member/res/MyPageRes.java
package historywowa.domain.member.presentation.dto.member.res;

import java.util.List;

public record MyPageRes(
        String userName,

        String profile,

        List<HeritageDto> heritageDtos
) {
    public static MyPageRes of(String userName, String profile, List<HeritageDto> heritageDtos) {
        return new MyPageRes(userName, profile, heritageDtos);
    }
}

File: src/main/kotlin/historywowa/domain/member/presentation/dto/member/res/MyPageRes.kt
package historywowa.domain.member.presentation.dto.member.res

data class MyPageRes(
        val userName: String,
        val profile: String?,
        val heritageDtos: List<HeritageDto>
) {
    companion object {
        fun of(userName: String, profile: String?, heritageDtos: List<HeritageDto>): MyPageRes {
            return MyPageRes(userName, profile, heritageDtos)
        }
    }
}
